### DISCUSSION TITLE
Better Swatch Attributes in OpenMage

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#4231] [Ideas]
This PR #3686 introduced a new feature to add colors to an attribute. At that time I showed that we cannot add colors to all attributes. For example, in the case of size attribute, you cannot add a color. Practically the only valid attribute that should have a color picker should have been one related to color. In the meantime, this aspect has been corrected but new challenges appeared.

The idea of ​​swatches is not a bad one, but the way of implementation should have been completely different, which would open up new possibilities and not be limited only to color. I propose to analyze what Magento 2 offers in this regard. Please check an online demo here

https://demo-magento-2.auroracreation.com/admin_panel/
Username: demo_admin
Password: demo_admin123

In the Backend go to **Stores > Attributes > Product** then add an attribute. Pay attention to "Catalog Input Type for Store Owner". Comparing with OpenMage it has more options, including **Visual Swatch** and **Text Swatch**. This is a the right implementation which guarantees more flexibility.

![type](https://github.com/user-attachments/assets/879b4781-8b58-4fed-9f59-923a3d0bf308)

For the **Visual Swatch** type, a color is entered manually or through the color-picker, an image is loaded.

![visual_swatch](https://github.com/user-attachments/assets/a3f77aae-c8b9-4168-882a-09bbf8661a5e)

For the **Text Swatch** type, enter a text and description.

![text_swatch](https://github.com/user-attachments/assets/81462202-f42c-4a64-ac7d-02b75510925b)

I opened this discussion for feedback.

**Comment by empiricompany** [comment]
The way it has been done is to be minimally invasive and not create problems with other extensions. 

The intention was not to implement a complete solution for the management of swatches but only to allow you to be able to create the color swatch automatically instead of upload individual images. 

I have also studied how it was implemented on Magento 2 and to replicate it in OpenMage there is a lot more effort. 

My clients instead are enthusiastic about it, They are finally independent in creating new colors without having to worry about creating an image and uploading it, like in other modern ecommerce platform, and without install (and integrate it in the theme) a new invasive extension to add this simple feature. 

Any way, it would be great to see a full implementation.

**Comment by sreichel** [comment]
Adding image upload should not be that difficult, but ...

... my focus is not on adding new features.

I wish there would be more peps doing reviews, adding issues ...

... then I would work on it.
